design/service-binding.md

```diff
@@ -0,0 +1,851 @@
+  annotations:
+    containerImage: docker.io/strimzi/operator:0.16.2
+spec:
+  customresourcedefinitions:
+    owned:
+    - description: Represents a user inside a Kafka cluster
+      displayName: Kafka User
+      kind: KafkaUser
+      name: kafkausers.kafka.strimzi.io
+      version: v1beta1
+      resources:
+      - kind: Secret
+        name: ''
+        version: v1
+      statusDescriptors:
+      - description: The username
+        displayName: Username
+        path: status.username
+        x-descriptors:
+        - binding:env:object:secret:username
+      - description: The secret containing the credentials
+        displayName: Secret
+        path: status.secret
+        x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:Secret
+        - binding:env:object:secret:password
+```
+
+### Consuming client's ServiceBindingRequest
+
+The binding needs to refer to the `Kafka` CR for the endpoint information and the `KafkaUser` CR for the username and password.
+
+``` yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBindingRequest
+metadata:
+  name: my-binding
+spec:
+  services:
+  - group: kafka.strimzi.io
+    kind: Kafka
+    version: v1beta1
+    resourceRef: my-cluster
+  - group: kafka.strimzi.io
+    kind: KafkaUser
+    version: v1beta1
+    resourceRef: my-user
+```
+
+There are of course two secrets now, the CA certificate secret accessed via the `Kafka` CR and the client's password secret accessed via the `KafkaUser` CR.
+
+The consuming client needs to know the listener name, the keys for the CA certificate secret fields for the certificate and password, and the key for the password field in the `KafkaUser` secret.
+
+The Service Binding Operator creates a `Secret` which contains:
+
+* **endpoints.<listener_name>** - bootstrap server information for all listeners
+* **ca.p12** - CA certificate PKCS #12 archive file for storing certificates and keys
+* **ca.password** - password for protecting the CA certficate PKCS #12 archive file
+* **username** - username for the consuming client
+* **password** - password for the consuming client
+
+
+## Binding to a Kafka cluster with mutual TLS authentication
+
+The `KafkaUser` CR should be sufficient to provide the information. For example:
+
+``` yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaUser
+metadata:
+  name: my-user
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  authentication:
+    type: tls
+  authorization:
+    type: simple
+    acls:
+      - resource:
+          type: topic
+          name: my-topic
+          patternType: literal
+        operation: Read
+status:
+  username: my-user-name
+  secret: my-user
+```
+
+The secret looks like this:
+
+``` yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-user
+  labels:
+    strimzi.io/kind: KafkaUser
+    strimzi.io/cluster: my-cluster
+type: Opaque
+data:
+  ca.crt: # Public key of the clients' CA
+  user.crt: # Public key of the user
+  user.key: # Private key of the user
```

Comment:
the Secret contains `user.p12` and `user.password` fields as well, which I see you are actually referring to in the following CSV so I would put them here as well for a better understanding of other readers.